index.js

```diff
@@ -20,39 +20,41 @@
 //. purpose: the [`typeof`][1] operator and [`Object.prototype.toString`][2].
 //. Each has pros and cons, but neither supports user-defined types.
 //.
-//. This package specifies an [algorithm][3] for deriving a _type identifier_
-//. from any JavaScript value, and exports an implementation of the algorithm.
-//. Authors of algebraic data types may follow this specification in order to
-//. make their data types compatible with the algorithm.
+//. sanctuary-type-identifiers comprises:
 //.
-//. ### Algorithm
+//.   - an npm and browser -compatible package for deriving the
+//.     _type identifier_ of a JavaScript value; and
+//.   - a specification which authors may follow to specify type identifiers
+//.     for their types.
 //.
-//. 1.  Take any JavaScript value `x`.
+//. ### Specification
 //.
-//. 2.  If `x` is `null` or `undefined`, go to step 6.
+//. For a type to be compatible with the algorithm:
 //.
-//. 3.  If `x.constructor` evaluates to `null` or `undefined`, go to step 6.
-//.
-//. 4.  If `x.constructor.prototype === x`, go to step 6. This check prevents a
-//.     prototype object from being considered a member of its associated type.
+//.   - every member of the type MUST have a `constructor` property pointing
+//.     to an object known as the _type representative_;
 //.
-//. 5.  If `typeof x.constructor['@@type']` evaluates to `'string'`, return
-//.     the value of `x.constructor['@@type']`.
+//.   - the type representative MUST have a `@@type` property
+//.     (the _type identifier_);
 //.
-//. 6.  Return the [`Object.prototype.toString`][2] representation of `x`
-//.     without the leading `'[object '` and trailing `']'`.
+//.   - the type identifier MUST be a string primitive and SHOULD have format
+//.     `'<namespace>/<name>[@<version>]'`, where:
 //.
-//. ### Compatibility
+//.       - The `namespace` MUST consist of one or more characters, and SHOULD
+//.         equal the name of the npm package which defines the type (including
+//.         [scope][4] where appropriate); and
 //.
-//. For an algebraic data type to be compatible with the [algorithm][3]:
+//.       - the `name` MUST consist of one or more characters, and SHOULD be
+//.         the unique name of the type; and
 //.
-//.   - every member of the type must have a `constructor` property pointing
-//.     to an object known as the _type representative_;
+//.       - the `version` MUST consist of one or more digits, and it SHOULD
+//.         represent the version of the type.
 //.
-//.   - the type representative must have a `@@type` property; and
+//.   - If the type identifier does not conform to the format specified above,
+//.     it is assumed that the entire string represents the _name_ of the type,
+//.     and _namespace_ will be `null`.
```

Comment:
I think we should make this change:
```diff
- and _namespace_ will be `null`
+ and _namespace_ and _versoin_ will be `null`
```

to make it clear that if `@@type` is `T@1` version will not be parsed